src/app/data-planes/components/PolicyTypeEntryList.vue

```diff
@@ -0,0 +1,161 @@
+<template>
+  <AccordionList
+    :initially-open="[]"
+    multiple-open
+  >
+    <AccordionItem
+      v-for="(policyEntry, index) in props.policyTypeEntries"
+      :key="index"
+    >
+      <template #accordion-header>
+        <h3>
+          <PolicyTypeTag :policy-type="policyEntry.type">
+            {{ policyEntry.type }} ({{ policyEntry.connections.length }})
+          </PolicyTypeTag>
+        </h3>
+      </template>
+
+      <template #accordion-content>
+        <div class="policy-list">
+          <KTable
+            class="policy-type-table"
+            :fetcher="() => ({ data: policyEntry.connections, total: policyEntry.connections.length })"
+            :headers="tableHeaders"
+            :cell-attrs="getCellAttributes"
+            disable-pagination
+            is-clickable
+          >
+            <template #sourceTags="{ rowValue }">
+              <TagList
+                v-if="Array.isArray(rowValue) && rowValue.length > 0"
+                class="tag-list"
+                :tags="rowValue"
+              />
+
+              <template v-else>
+                —
+              </template>
+            </template>
+
+            <template #destinationTags="{ rowValue }">
+              <TagList
+                v-if="Array.isArray(rowValue) && rowValue.length > 0"
+                class="tag-list"
+                :tags="rowValue"
+              />
+
+              <template v-else>
+                —
+              </template>
+            </template>
+
+            <template #name="{ rowValue }">
+              <template v-if="rowValue !== null">
+                {{ rowValue }}
+              </template>
+
+              <template v-else>
+                —
+              </template>
+            </template>
+
+            <template #origins="{ rowValue }">
+              <div
+                v-if="rowValue.length > 0"
+                class="origin-list"
+              >
+                <router-link
+                  v-for="(origin, originIndex) in rowValue"
+                  :key="`${index}-${originIndex}`"
+                  :to="origin.route"
+                >
+                  {{ origin.name }}
+                </router-link>
+              </div>
+
+              <template v-else>
+                —
+              </template>
+            </template>
+
+            <template #config="{ rowValue, rowKey }">
+              <template v-if="rowValue !== null">
+                <CodeBlock
+                  :id="`${index}-${rowKey}-code-block`"
```

Comment:
Because the component contains elements that have `for`/`id` attribute relationships that must be unique within the document. While adding the component to Kongponents, we decided to make the `id` mandatory so one can’t forget to explicitly take care of it.

Reply:
Oh I see, for the code searching form input? In this case that means there can only ever be one instance of this particular `PolicyTypeEntryList` component in the page at once? 

Do you know if Vue has its own recommended native uuid generator, I had a very brief look but didn't see one see unless I missed it? Sorry getting a bit off topic, but just getting a feel for things still.

Reply:
> Oh I see, for the code searching form input?

Yes

> In this case that means there can only ever be one instance of this particular PolicyTypeEntryList component in the page at once?

I’m not sure what you mean by this, but I did notice an issue with the IDs here. Since this component is used twice on this page, I actually need to make these IDs more unique. Fixed now.

> Do you know if Vue has its own recommended native uuid generator, I had a very brief look but didn't see one see unless I missed it? Sorry getting a bit off topic, but just getting a feel for things still.

Don’t think so, but there are utilities for doing this (Kongponents uses one).

Reply:
> In this case that means there can only ever be one instance of this particular PolicyTypeEntryList component in the page at once?

>> I’m not sure what you mean by this, but I did notice an issue with the IDs here. Since this component is used twice on this page, I actually need to make these IDs more unique. Fixed now.

Ah! I didn't even notice this was on this page twice 😄 , but yeah exactly that is what I meant.

> (Kongponents uses one).

Do you know whereabouts that is? I had a quick look but didn't see it

Reply:
Any component that imports the following (e.g. KTable, 10 in total):

```js
import { v1 as uuidv1 } from 'uuid'
```

Reply:
Oh I see, hmm not quite what I was thinking but thanks, good to know